packages/contracts-bedrock/contracts/L1/ResourceMetering.sol

```diff
@@ -134,7 +134,14 @@ abstract contract ResourceMetering is Initializable {
         );
 
         // Determine the amount of ETH to be paid.
-        uint256 resourceCost = _amount * params.prevBaseFee;
+        // Safety: _amount is a uint64
+        //         params.prevBaseFee is a uint128
+        //         resourceCost is a uint256
+        //         type(uint64).max * type(uint128).max < type(uint256).max
+        uint256 resourceCost;
+        unchecked {
+            resourceCost = _amount * params.prevBaseFee;
+        }
```

Comment:
Still curious, why did it then throw an overflow error? Is it because `_amount` is a `uint64` and `params.prevBaseFee` is a `uint128` and the solc performs the calculation `_amount * params.prevBaseFee` first in `uint128` and only then converts into the target type `uint256`? Because in that case we'd need to do
```suggestion
        uint256 resourceCost = uint256(_amount) * uint256(params.prevBaseFee);
```
instead. But I don't know the solc that well, so no idea :)

Reply:
Do we have tests that would fail in case the overflow calculation would indeed only happen in `uint128`?

Reply:
Its possible to do the casting, but there is no real need to do so because it will result in checked math which will bloat the bytecode and cause it to use slightly more gas. This is why I opted into using an unchecked block, since we have a guarantee that there will not be an overflow

Re: tests, I do have another commit locally on top of this commit that I am working on to thoroughly test to ensure that all possible values can be represented, I broke this change out into its own commit so that it would be easier to review on its own. It is reproducible in chisel. I'll see if i can make a unit test for this specifically without a large diff